### PR TITLE
Main packaging fixups

### PR DESCRIPTION
**Pull request by egernst**
Our release scripting and packaging scripts had dependencies on obs-packaging, which was removed. Pulled those scripts out and added some cleanup. This PR:

* Reintroduces gen_versions_txt.sh, and build_image.sh
* Cleans up some of the packaging scripts to take new paths into account, and for clean handling.
* Updates GitHub workflows to take new versions_txt path into account.
* Introduces the kata-deploy GitHub workflow, for testing kata-deploy script changes in the kata-containers/kata-containers repo.

### CODE REVIEW

File: tools/packaging/release/kata-deploy-binaries.sh [hunk 1]
```diff
@@ -273,14 +275,14 @@ main() {
...
 	mkdir -p "${destdir}"
-	install_image
 	install_kata_components
 	install_experimental_kernel
 	install_kernel
 	install_clh
 	install_qemu
 	install_qemu_virtiofsd
 	install_firecracker
+	install_image
```
Comment (amshinde) [inline tools/packaging/release/kata-deploy-binaries.sh:285]: I see you changed the order here. Any particular reason?

File: tools/packaging/Makefile [hunk 2]
```diff
@@ -18,20 +18,10 @@ export SNAPCRAFT_FILE
 export VERSION_FILE
 export VERSIONS_YAML_FILE
 
-test:
-	@$(MK_DIR)/.ci/test.sh
```
Comment (amshinde) [inline tools/packaging/Makefile:22]: Is this no longer used?

File: .github/workflows/kata-deploy-test.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
+on: issue_comment
+name: test-kata-deploy
```
Comment (amshinde) [inline .github/workflows/kata-deploy-test.yaml:2]: On similar lines...Was just checking our existing release workflow and I see that it is still using the action from `https://github.com/kata-containers/packaging` : 
https://github.com/kata-containers/kata-containers/blob/ceebd06b64f086c21b4c5ee6fd7fb594559af5c1/.github/workflows/main.yaml#L297

We should fix that to use an action in this repo as well.

File: tools/packaging/Makefile (hunk 2 above)
Comment (egernst) [reply tools/packaging/Makefile:22]: Not sure if it ever was.  test.sh doesn't exist in this repository.

File: tools/packaging/release/kata-deploy-binaries.sh (hunk 1 above)
Comment (egernst) [reply tools/packaging/release/kata-deploy-binaries.sh:285]: no, image took too long when testing the others.

File: .github/workflows/kata-deploy-test.yaml (hunk 3 above)
Comment (egernst) [reply .github/workflows/kata-deploy-test.yaml:2]: That is for 1.x. release.yaml is for 2.x

Comment (amshinde) [reply .github/workflows/kata-deploy-test.yaml:2]: got it! The confusion while switching from 1.x to 2.x continues..:)

File: tools/packaging/Makefile (hunk 2 above)
Comment (amshinde) [reply tools/packaging/Makefile:22]: Ok, this must have been used while running the CI for 1.x packaging repo.

**Review by amshinde — APPROVED** [review]

**Review by likebreath — APPROVED** [review]

**Pull request merged**